acvm/src/lib.rs

```diff
@@ -92,7 +92,7 @@ fn first_missing_assignment(
     })
 }
 
-pub trait Backend: SmartContract + ProofSystemCompiler + PartialWitnessGenerator {}
+pub trait Backend: SmartContract + ProofSystemCompiler + PartialWitnessGenerator + Default {}
```

Comment:
Does default allow us to do some setup? I think some backends probably need to do setup work in the function so maybe this should be a `new() -> Self` function if default doesn't allow setup work

Reply:
What sort of setup would you want to do that can't be done in `default()`? afaik they're directly equivalent.

My choice for `Default` over `new() -> Self` is that:
a. it's a standardized trait whereas `new()` is an ad hoc method.
b. we can let backends define their own `new` function which takes additional methods for e.g. injecting a CRS or something.